File: EasyDelivery.Application/DTOs/Pedido/PedidoRequest.cs
using EasyDelivery.Application.DTOs.ItemPedido;
using EasyDelivery.Domain.Entities;
using EasyDelivery.Domain.Entities.Enums;

namespace EasyDelivery.Application.DTOs.Pedido
{
    public class PedidoRequest
    {
        public int Id { get; set; }
        public int ClienteId { get; set; }
        public int RestauranteId { get; set; }
        public int? EntregadorId { get; set; }
        public DateTime DataCriacao { get; set; }
        public DateTime? HoraSaida { get; set; }
        public DateTime? HoraEntrega { get; set; }
        public StatusPedido Status { get; set; }

        public List<ItemPedidoResponse> Itens { get; set; } = new List<ItemPedidoResponse>();
    }
}

File: easydelivery-front/src/app/models/pedido/pedidoRequest.ts
import { StatusPedido } from "../enums/statusPedido";
import { ItensPedidoRequest } from "./itensPedidoRequest";

export interface PedidoRequest{
    id?: number | null;
    clienteId: number;
    restauranteId: number;
    entregadorId: number | null;
    dataCriacao: Date;
    horaSaida: Date | null;
    horaEntrega: Date | null;
    status: StatusPedido;
    itens: ItensPedidoRequest[];
}

